src/Compilers/CSharp/Portable/Symbols/PublicModel/Symbol.cs

```diff
@@ -98,7 +99,9 @@ bool ISymbol.Equals(ISymbol other, CodeAnalysis.SymbolEqualityComparer equalityC
 
         protected bool Equals(Symbol other, CodeAnalysis.SymbolEqualityComparer equalityComparer)
         {
-            return other is object && UnderlyingSymbol.Equals(other.UnderlyingSymbol, equalityComparer.CompareKind);
+            var areEqual = other is object && UnderlyingSymbol.Equals(other.UnderlyingSymbol, equalityComparer.CompareKind);
+            Debug.Assert(!areEqual || this.GetHashCode() == other.GetHashCode(), "Hash code should be the same for equal symbols.");
```

Comment:
>Debug.Assert(!areEqual || this.GetHashCode() == other.GetHashCode(), "Hash code should be the same for equal symbols."); [](http://example.com/codeflow?start=12&length=120)

Is this going to duplicate check performed by SymbolEqualityComparer?

Reply:
@AlekseyTs Yep. Some callers go into a single path of the two.